BUILDING_CALICO.md

```diff
@@ -0,0 +1,300 @@
+# Building Calico components
+
+This document describes how to build the following Calico components from their
+source code:
+
+-  felix
+-  typha
+-  CNI plugins
+-  confd (Calico specific fork)
+-  bird and bird6 (Calico specific fork)
+-  Calico BGP daemon
+-  calicoctl
+-  calico/node
+
+## Requirements
+
+These build instructions assume you have a Linux build environment
+with:
+
+-  Docker 1.12+
+-  git
+-  make
+
+#### Build environment
+
+We recommend checking out all of the code to a common base directory.  The instructions
+below assume the environment variable $BASEDIR has been set.  For example:
+
+```
+export BASEDIR=~/go/src/github.com/projectcalico
+```
+
+This directory should exist:
+
+```
+mkdir -p $BASEDIR
+```
+
+These instructions describe how to build a particular release or Calico.  This comprises
+of multiple sub components each indivudally versioned.  To determine which
+particular tag (version) of code to checkout for each repo, consult the Releases page of the
+appropriate version of the [Calico documentation](http://docs.projectcalico.org)
+
+Define the following environment variables that describe the versions of each of the
+components that you are building (the instruction below use these):
+
+```
+VERSION_TYPHA
+VERSION_FELIX
+VERSION_CNI
+VERSION_CONFD
+VERSION_BIRD
+VERSION_BGP_DAEMON
+VERSION_LIBNETWORK
+VERSION_CALICOCTL
+```
+
+For example, the v2.3.0 release of Calico would define the following:
+
+```
+export VERSION_TYPHA=v0.2.0
+export VERSION_FELIX=2.3.0-rc3
+export VERSION_CNI=v1.9.1
+export VERSION_CONFD=v0.12.1-calico0.1.0
+export VERSION_BIRD=v0.3.1
+export VERSION_BGP_DAEMON=v0.2.1
+export VERSION_LIBNETWORK=v1.1.0
+export VERSION_CALICOCTL=v1.3.0-rc1
+```
+
+## Build instructions
+
+The following sections should be followed in order as there are cross dependencies
+between the various components that have been built.
+
+### 1. typha
+
+To build typha, clone the typha repo and checkout the correct tag.
+
+```
+cd $BASEDIR
+git clone git@github.com:projectcalico/typha.git
+cd typha
+git checkout tags/$VERSION_TYPHA
+```
+
+Build the `calico/typha` docker image (which also builds the associated binaries) and
+retag the image with the correct version.
+
+```
+make clean
+make calico/typha
+docker tag calico/typha calico/typha:$VERSION_TYPHA
+```
+
+#### What has this built?
```

Comment:
Yeah we can do, and since some of the artifacts have not been built it might be a better heading.